File: Provas/2016.1/questao1-java/Decorator.java
public class Decorator extends IComponent
{
    public Decorator() {
        this.decorated = null;
    }
    public Decorator(IComponent decorated) {
        setDecorated(decorated);
    }
    public void setDecorated(IComponent decorated) {
        this.decorated = decorated;    
    }
    public void aumentarPreco(double percentual) {
        decorated.aumentarPreco(percentual);
    }
    IComponent decorated;
}

File: Provas/2016.1/questao1-qt/decorator.cpp
#include "decorator.h"

#include "icomponent.h"

Decorator::Decorator(IComponent *decorated) {
    setDecorated(decorated);
}

void Decorator::setDecorated(IComponent *decorated) {
    m_decorated = decorated;
}

void Decorator::aumentarPreco(double percentual) {
    m_decorated->aumentarPreco(percentual);
}
